src/ip/Address.cc

```diff
@@ -670,7 +670,7 @@ Ip::Address::getAddrInfo(struct addrinfo *&dst, int force) const
 
     } else if ( force == AF_INET || (force == AF_UNSPEC && isIPv4()) ) {
 
-        dst->ai_addr = (struct sockaddr*)new sockaddr_in;
+        dst->ai_addr = (struct sockaddr*)new sockaddr_storage;
```

Comment:
Please replace this with a call to `InitAddr` to minimize the places doing new/delete.
```suggestion
        InitAddr(dst);
```

(or better, place this above the per-family if-statement, so only done once for all of them.)